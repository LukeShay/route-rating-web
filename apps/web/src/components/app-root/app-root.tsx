import { Component, ComponentInterface, h } from '@stencil/core'

@Component({
  tag: 'app-root',
  styleUrl: 'app-root.css',
  shadow: true,
})
export class Root implements ComponentInterface {
  render() {
    return (
      <div>
        <app-navigation />

        <main>
          <stencil-router>
            <stencil-route-switch scrollTopOffset={0}>
              <stencil-route url="/" component="app-home" exact={true} />
              <stencil-route url="/profile" component="app-profile" />
            </stencil-route-switch>
          </stencil-router>
        </main>
      </div>
    )
  }
}
