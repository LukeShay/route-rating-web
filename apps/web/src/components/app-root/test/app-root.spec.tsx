import { Root } from '../app-root'
import { newSpecPage } from '@stencil/core/testing'

describe('app-profile', () => {
  it('should match snapshot', async () => {
    const page = await newSpecPage({
      components: [Root],
      html: `<app-root></app-root>`,
    })
    expect(page.root).toMatchSnapshot()
  })
})
