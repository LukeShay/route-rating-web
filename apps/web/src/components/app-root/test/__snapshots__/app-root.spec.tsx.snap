// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`app-profile should match snapshot 1`] = `
<app-root>
  <mock:shadow-root>
    <div>
      <app-navigation></app-navigation>
      <main>
        <stencil-router>
          <stencil-route-switch scrolltopoffset="0">
            <stencil-route component="app-home" exact="" url="/"></stencil-route>
            <stencil-route component="app-profile" url="/profile"></stencil-route>
          </stencil-route-switch>
        </stencil-router>
      </main>
    </div>
  </mock:shadow-root>
</app-root>
`;
